src/core/zip.ts

```diff
@@ -81,3 +72,49 @@ export async function zip(config?: InlineConfig): Promise<string[]> {
 
   return zipFiles;
 }
+async function zipDir(
+  directory: string,
+  outputPath: string,
+  options?: {
+    include?: string[];
+    exclude?: string[];
+    transform?: (
+      file: string,
+      content: string,
+    ) => Promise<string | undefined | void> | string | undefined | void;
+    additionalWork?: (archive: JSZip) => Promise<void> | void;
```

Comment:
Specifically I've added the `transform` option to transform the file before zipping, and the `additionalWork` option to add generated files to the zip.